addons/account_peppol/models/account_journal.py

```diff
@@ -7,28 +7,62 @@ class AccountJournal(models.Model):
     account_peppol_proxy_state = fields.Selection(related='company_id.account_peppol_proxy_state')
     is_peppol_journal = fields.Boolean(string="Account used for Peppol", default=False)
 
-    def peppol_get_new_documents(self):
+    def _peppol_import_invoice(self, attachment, partner_endpoint, peppol_state, uuid):
+        """Save new documents in an accounting journal, when one is specified on the company.
+
+        :param attachment: the new document
+        :param partner_endpoint: DEPRECATED - to be removed in master
+        :param peppol_state: the state of the received Peppol document
+        :param uuid: the UUID of the Peppol document
+        :return: the created move
+        """
+
+        move = self.env['account.move'].create({
+            'journal_id': self.id,
+            'move_type': 'in_invoice',
+            'peppol_move_state': peppol_state,
+            'peppol_message_uuid': uuid,
+        })
+        if 'is_in_extractable_state' in move._fields:
+            move.is_in_extractable_state = False
+
+        move._extend_with_attachments(attachment, new=True)
+        move._message_log(
+            body=_(
+                "Peppol document (UUID: %(uuid)s) has been received successfully",
+                uuid=uuid,
+            ),
+            attachment_ids=attachment.ids,
+        )
+        attachment.write({'res_model': 'account.move', 'res_id': move.id})
+        return move
+
+    def button_fetch_einvoices(self):
+        super().button_fetch_einvoices()
         edi_users = self.env['account_edi_proxy_client.user'].search([
             ('company_id.account_peppol_proxy_state', '=', 'receiver'),
             ('company_id', 'in', self.company_id.ids),
         ])
         edi_users._peppol_get_new_documents()
 
-    def peppol_get_message_status(self):
+    def button_fetch_einvoices_status(self):
+        super().button_fetch_einvoices_status()
         can_send = self.env['account_edi_proxy_client.user']._get_can_send_domain()
         edi_users = self.env['account_edi_proxy_client.user'].search([
             ('company_id.account_peppol_proxy_state', 'in', can_send),
             ('company_id', 'in', self.company_id.ids),
         ])
         edi_users._peppol_get_message_status()
 
-    def action_peppol_ready_moves(self):
-        return {
-            'name': _("Peppol Ready invoices"),
-            'type': 'ir.actions.act_window',
-            'view_mode': 'list,form',
-            'res_model': 'account.move',
-            'context': {
-                'search_default_peppol_ready': 1,
-            }
-        }
+    @api.depends('account_peppol_proxy_state')
+    def _compute_show_fetch_einvoices_status_button(self):
```

Comment:
Compute methods should be above i the file.
![image](https://github.com/user-attachments/assets/d88deb67-3710-404f-84d5-746c382b524a)
https://www.odoo.com/documentation/master/contributing/development/coding_guidelines.html#symbols-and-conventions